packages/libs/user-datasets/src/lib/Components/Sharing/UserDatasetSharingModal.jsx

```diff
@@ -460,8 +469,35 @@ class UserDatasetSharingModal extends React.Component {
         </div>
       );
     } else {
+      const visibilities = uniq(datasets.map((d) => d.meta.visibility));
+      const visibility =
+        visibilities.length === 1 ? visibilities[0] : undefined;
```

Comment:
Nit: add a comment here?

This is clever, but it took me a bit of effort to understand the how and why of `visibilities` and `visibility`, and how it works to determine the value of the toggle.

Reply:
Ah, yes, I will add a comment. Thanks!

Reply:
Comment added. Let me know what you think.